src/mscorlib/src/System/Lazy.cs

```diff
@@ -322,147 +528,16 @@ public T Value
         {
             get
             {
-                Boxed boxed = null;
-                if (m_boxed != null )
+                while (true)
```

Comment:
Or it can be - without changing `CreateValue` to return the value:

```
if (_state != null)
     CreateValue();
return _value;
```

This only probably slightly better because it has smaller IL, and thus it will make it better inlining candidate.